lighthergm/R/gof_lighthergm.R

```diff
@@ -151,40 +152,69 @@ gof_lighthergm <- function(net,
                            colname_vertex_id,
                            colname_block_membership,
                            lighthergm_results,
+                           type = 'full',
                            ergm_control = ergm::control.simulate.formula(),
                            seed = NULL,
                            n_sim = 1,
                            prevent_duplicate = TRUE,
                            compute_geodesic_distance = FALSE,
                            ...) {
-  gof_formula <- swap_formula_network(net, lighthergm_results$est_within$formula, environment())
 
+  # Setup
+  gof_formula <- swap_formula_network(net, lighthergm_results$est_within$formula, environment())
+  burnin <- ergm_control$MCMC.burnin
+  interval <- ergm_control$MCMC.interval
   coef_within_block <- coef(lighthergm_results$est_within)
   coef_between_block <- coef(lighthergm_results$est_between)
 
-  original_stats <- get_gof_stats(gof_formula, compute_geodesic_distance = compute_geodesic_distance)
+  # Validate the simulation type
+  allowed_type_values <- c('full', 'within')
+  if (!type %in% allowed_type_values){
+    stop('The `type` argument must be any of `full` or `within`')
```

Comment:
Instead of using back quotes for full and within, this might be slightly better:
```suggestion
    stop("The `type` argument must be any of 'full' or 'within'")
```